design/custom-request-timeout.md

```diff
@@ -0,0 +1,256 @@
+# Executive Summary
+
+**Status**: _Draft_
+
+This document describes the design of a new resource in IngressRoute for custom request timeout and retries. 
+
+This new resource will be integrated into Contour 0.11.
+
+# Goals
+
+- Generic resource name for configuring custom request timeouts per IngressRoute
+- Generic resource name for configuring custom retry attempts per IngressRoute
+
+# Non-goals
+
+- None
+
+# Background
+
+Contour supports custom request timeout and custom retry attempts via Ingress. The same functionality has been
+requested via IngressRoute as well.
```

Comment:
Please link to the documentation or issue for Ingress so its clear to the reader what we're replicating in IngressRoute.